helios-services/src/main/java/com/spotify/helios/agent/EnvironmentVariableReporter.java

```diff
@@ -33,28 +33,26 @@
  * Puts the environment variables the Agent has been configured to set in all child containers
  * into ZK so they can be visible to the master and via the API.
  */
-public class EnvironmentVariableReporter extends InterruptingScheduledService {
+public class EnvironmentVariableReporter extends SignalAwaitingService {
 
   private static final int RETRY_INTERVAL_MILLIS = 1000;
 
   private final Map<String, String> envVars;
   private final ZooKeeperNodeUpdater nodeUpdater;
-  private final CountDownLatch zkRegistrationSignal;
 
   public EnvironmentVariableReporter(final String host, final Map<String, String> envVars,
                                      final NodeUpdaterFactory nodeUpdaterFactory,
                                      final CountDownLatch zkRegistrationSignal) {
+    super(zkRegistrationSignal);
```

Comment:
just inconsistencies from adding it in some places versus modifying existing code. I'll update